### ISSUE TITLE
[import/default]: false negative when import CJS module

### ISSUE THREAD

**Issue by tpluscode** [import-js/eslint-plugin-import#2769]
I found a weird problem with the `import/default` rule. See [this minimal gist](https://gist.github.com/tpluscode/50d638600e4397a0bef4b3289cfb2925)

I import two modules which are very similar, both being CJS with default export.

```js
import ecmaScriptLiteralLoader from 'rdf-loader-code/ecmaScriptLiteral.js'
import ecmaScriptModuleLoader from 'rdf-loader-code/ecmaScriptModule.js'
```

The first is fine but the second triggers a rule

```
3:8  error  No default export found in imported module "rdf-loader-code/ecmaScriptModule.js"  import/default
```

**Comment by tpluscode** [comment]
Right here:

https://github.com/zazuko/rdf-loader-code/blob/master/ecmaScriptLiteral.js
https://github.com/zazuko/rdf-loader-code/blob/master/ecmaScriptModule.js

Gists do not support directories so I moved that repro to a normal repo: https://github.com/tpluscode/import-default

**Comment by ljharb** [comment]
I'm pretty sure they do, they're just git repos - maybe put a slash in the filename? but a repo's fine too :-)

I'm not sure why you declare the function and then assign it rather than `module.exports = async function loader`, but it should still work fine.

Given that the one that works is a function and the one that doesn't is an async function, I assume that the detection code needs to check for the differing kind of AST node type.

**Comment by ljharb** [comment]
Although, the `default` rule only checks ESM, not CJS - so it's confusing that the first one wouldn't error. If you comment out the second one, does the first one error?

**Comment by tpluscode** [comment]
> I'm not sure why you declare the function and then assign it rather than

To be able to set `loader.register = ...`

I tried a bunch of things and nothing made a difference until I [removed the dynamic `import`](https://github.com/tpluscode/import-default/actions/runs/4827846978/jobs/8601000350)
